src/main/java/org/zalando/riptide/RestBase.java

```diff
@@ -0,0 +1,61 @@
+package org.zalando.riptide;
+
+/*
+ * ⁣​
+ * Riptide
+ * ⁣⁣
+ * Copyright (C) 2015 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import java.net.URI;
+import java.util.Map;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.util.UriTemplateHandler;
+
+abstract class RestBase<D> {
```

Comment:
Side note: This would be a lot easier if our two different Dispatchers would share a common interface, but unfortunately the dispatch method uses a generic array and that is only safe (no compiler warnings) with a @SafeVarargs annotation, which is not possible to be placed on an interface. We could make it into an abstract class, maybe?